launchable/utils/http_client.py

```diff
@@ -28,9 +33,25 @@ def request(self, method, path, **kwargs):
 
         logger = Logger()
         try:
-            logger.audit(
-                "send request method:{} path:{} headers:{} args:{}".format(method, path, headers, kwargs))
-            response = self.session.request(method, url, headers={**headers, **self._headers()}, **kwargs)
+            if LOG_LEVEL_AUDIT >= logging.root.level:
+                log_kwargs = copy.copy(kwargs)
+                if any("gzip" in h for h in headers.values()) and "data" in log_kwargs:
+                    data = log_kwargs.pop("data")
+                    if isinstance(data, types.GeneratorType):
+                        generator, _generator = itertools.tee(data)
```

Comment:
https://docs.python.org/3/library/itertools.html#itertools.tee

> Once tee() has made a split, the original iterable should not be used anywhere else; otherwise, the iterable could get advanced without the tee objects being informed.